provider/mesos.go

```diff
@@ -0,0 +1,449 @@
+		return "0"
+	}
+
+	if label, err := provider.getLabel(application, "traefik.weight"); err == nil {
+		return label
+	}
+	return "0"
+}
+
+func (provider *Mesos) getDomain(task state.Task) string {
+	if label, err := provider.getLabel(task, "traefik.domain"); err == nil {
+		return label
+	}
+	return provider.Domain
+}
+
+func (provider *Mesos) getProtocol(task state.Task, applications []state.Task) string {
+	application, errApp := getMesos(task, applications)
+	if errApp != nil {
+		log.Errorf("Unable to get mesos application from task %s", task.DiscoveryInfo.Name)
+		return "http"
+	}
+	if label, err := provider.getLabel(application, "traefik.protocol"); err == nil {
+		return label
+	}
+	return "http"
+}
+
+func (provider *Mesos) getPassHostHeader(task state.Task) string {
+	if passHostHeader, err := provider.getLabel(task, "traefik.frontend.passHostHeader"); err == nil {
+		return passHostHeader
+	}
+	return "false"
+}
+
+func (provider *Mesos) getPriority(task state.Task) string {
+	if priority, err := provider.getLabel(task, "traefik.frontend.priority"); err == nil {
+		return priority
+	}
+	return "0"
+}
+
+func (provider *Mesos) getEntryPoints(task state.Task) []string {
+	if entryPoints, err := provider.getLabel(task, "traefik.frontend.entryPoints"); err == nil {
+		return strings.Split(entryPoints, ",")
+	}
+	return []string{}
+}
+
+// getFrontendRule returns the frontend rule for the specified application, using
+// it's label. It returns a default one (Host) if the label is not present.
+func (provider *Mesos) getFrontendRule(task state.Task) string {
+	// ⚠⚠⚠⚠⚠⚠⚠⚠⚠⚠⚠⚠⚠⚠⚠⚠⚠⚠⚠⚠⚠⚠⚠⚠⚠⚠⚠⚠⚠⚠⚠⚠⚠⚠⚠⚠⚠⚠
+	// TODO: backwards compatibility with DEPRECATED rule.Value
+	if value, err := provider.getLabel(task, "traefik.frontend.value"); err == nil {
+		log.Warnf("Label traefik.frontend.value=%s is DEPRECATED, please refer to the rule label: https://github.com/containous/traefik/blob/master/docs/index.md#mesos", value)
+		rule, _ := provider.getLabel(task, "traefik.frontend.rule")
+		return rule + ":" + value
+	}
+	// ⚠⚠⚠⚠⚠⚠⚠⚠⚠⚠⚠⚠⚠⚠⚠⚠⚠⚠⚠⚠⚠⚠⚠⚠⚠⚠⚠⚠⚠⚠⚠⚠⚠⚠⚠⚠⚠⚠
+	if label, err := provider.getLabel(task, "traefik.frontend.rule"); err == nil {
+		return label
+	}
+	return "Host:" + strings.ToLower(strings.Replace(provider.getSubDomain(task.DiscoveryInfo.Name), "_", "-", -1)) + "." + provider.Domain
+}
+
+func (provider *Mesos) getBackend(task state.Task, applications []state.Task) string {
+	application, errApp := getMesos(task, applications)
+	if errApp != nil {
+		log.Errorf("Unable to get mesos application from task %s", task.DiscoveryInfo.Name)
+		return ""
+	}
+	return provider.getFrontendBackend(application)
+}
+
+func (provider *Mesos) getFrontendBackend(task state.Task) string {
+	if label, err := provider.getLabel(task, "traefik.backend"); err == nil {
+		return label
+	}
+	return "-" + strings.Replace(strings.Replace(task.DiscoveryInfo.Name, ".", "-", -1), "_", "-", -1)
```

Comment:
As `strings.Replace(strings.Replace(task.DiscoveryInfo.Name, ".", "-", -1), "_", "-", -1)` is called multiple times, it would be great to create a function :)